spec/repositories/marshallers/section_association_marshaller_spec.rb

```diff
@@ -64,16 +57,22 @@
     end
 
     it "decorates the entity with the attributes" do
+      allow(ManualWithSections).to receive(:new)
+      allow(SectionBuilder).to receive(:new).and_return(:section_builder)
+      allow(NullValidator).to receive(:new)
+      allow(ManualValidator).to receive(:new)
+
       marshaller.load(entity, record)
 
-      expect(decorator).to have_received(:call).
-        with(entity, sections: sections, removed_sections: removed_sections)
+      expect(ManualWithSections).to have_received(:new).with(:section_builder, entity, sections: [section], removed_sections: [removed_section])
+      expect(NullValidator).to have_received(:new)
+      expect(ManualValidator).to have_received(:new)
```

Comment:
I'm not convinced this is the best way to test this - I find it very difficult to follow.

Although it might introduce some duplication, I think it might be worth experimenting with testing the emergent behaviour of the chain of delegators, rather than attempting any mocking at all. Let me know if this doesn't make sense and I can try to explain in Slack or in person tomorrow.